content/java/concepts/errors/terms/typenotpresentexception/typenotpresentexception.md

```diff
@@ -0,0 +1,56 @@
+---
+Title: 'TypeNotPresentException'
+Description: 'Occurs when an application tries to access a type using a string representing the name of the type, but no definition for the type with the specified name can be found.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Errors'
+  - 'Error Handling'
+  - 'Exceptions'
+  - 'Debugging'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`TypeNotPresentException`** is an exception in Java that occurs when an application tries to access a type using its name as a string, but the type definition cannot be found. Unlike ClassNotFoundException, this exception is unchecked. It can happen when attempting to access undefined type variables or when loading classes, interfaces, or annotation types. This exception is particularly relevant when using reflective annotation reading APIs. Overall, the TypeNotPresentException is thrown when the specified type is not present or cannot be accessed at runtime.
+
+## Example
+
+In this below code, we have a generic class `TypeNotPresentExceptionExample` with a type parameter T. We attempt to instantiate the class with the type argument NonExistentClass, which is a non-existent class. This triggers a `TypeNotPresentException` since the type NonExistentClass is not available at runtime We catch the exception and display the type name and cause.
```

Comment:
[Nitpick] Pronouns are not allowed to be used according to the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#writing-entries), so consider revising. 
```suggestion
In this below code, we have a generic class `TypeNotPresentExceptionExample` with a type parameter T. We attempt to instantiate the class with the type argument NonExistentClass, which is a non-existent class. This triggers a `TypeNotPresentException` since the type NonExistentClass is not available at runtime We catch the exception and display the type name and cause.
```